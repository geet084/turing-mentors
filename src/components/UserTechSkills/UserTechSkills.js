import React, { Component } from 'react';
import { CreateCheckbox } from '../CreateCheckbox/CreateCheckbox';

export class UserTechSkills extends Component {
  constructor(props) {
    super(props)
    const { tech_skills } = this.props

    this.state = {
      tech_skills: tech_skills || [],
    }
  }

  submitForm = (e) => {
    e.preventDefault();
    this.props.updateUserInfo([this.state, 'userNonTechSkills']);
  }

  goBack = (e) => {
    e.preventDefault();
    this.props.updateUserInfo([this.state, 'userSchedule']);
  }

  checkBoxes = ({ target }) => {
    let skills = [...this.state.tech_skills]
    let updatedState;

    if (skills.includes(target.value)) updatedState = skills.filter(skill => skill !== target.value)
    else updatedState = [...skills, target.value]

    this.setState({ tech_skills: updatedState })
  }

  render() {
    const mentor = this.props.user === 'Mentor';
    const tech_skills = ['ruby', 'javascript', 'python', 'java', 'elixer', 'c', 'php', 'swift', 'sql']
    const checkBoxes = tech_skills.map((skill, index) => {
      return <CreateCheckbox key={skill} field={skill} name={skill} value={index + 1} checkBoxes={this.checkBoxes} checked={this.state.tech_skills.includes(index.toString()) ? true : false} />
    })

    return (
      <div className="form-page">
        <p className="pages">Mentor Tech skills</p>
        <form>
          {checkBoxes}
        </form>
        <button className="next-btn" onClick={this.goBack}>Back</button>
        <button className="next-btn" onClick={this.submitForm}>Next</button>
        <span className="pages">5 of {mentor ? '6' : '4'}</span>
      </div>
    )
  }
}

export default UserTechSkills;