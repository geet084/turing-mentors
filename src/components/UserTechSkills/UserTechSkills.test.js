import React from 'react';
import { shallow } from 'enzyme';
import { UserTechSkills } from './UserTechSkills';

describe('UserTechSkills', () => {
  let wrapper;
  const mockProps = {
    updateUserInfo: jest.fn()
  };
  const mockState = {
    tech_skills: []
  };

  beforeEach(() => {
    wrapper = shallow(<UserTechSkills {...mockProps} />);
  });

  describe('initial snapshot', () => {
    it('should match the correct default snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the correct populated non_tech_skills snapshot', () => {
      wrapper.setState({ tech_skills: ['1'] });
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the correct user snapshot', () => {
      wrapper.setProps({ user: 'Mentor' });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('submitForm', () => {
    it('should handle the form submission', () => {
      const mockEvent = { preventDefault: () => { } };
      const expected = [mockState, 'userNonTechSkills'];

      wrapper.instance().submitForm(mockEvent);

      expect(mockProps.updateUserInfo).toHaveBeenCalledWith(expected);
    });
  });

  describe('goBack', () => {
    it('should handle going back a page', () => {
      const mockEvent = { preventDefault: () => { } };
      const expected = [mockState, 'userSchedule'];

      wrapper.instance().goBack(mockEvent);

      expect(mockProps.updateUserInfo).toHaveBeenCalledWith(expected);
    });
  });

  describe('checkBoxes', () => {
    it('should update the selected input in state', () => {
      const mockEvent = { target: { value: 1 } };
      const expected = {
        tech_skills: [1]
      }

      wrapper.instance().checkBoxes(mockEvent);
      expect(wrapper.state()).toEqual(expected);
    });

    it('should update the deselected input in state', () => {
      const mockEvent = { target: { value: 3 } };
      const changed = { tech_skills: [3] };
      const expected = { tech_skills: [] };

      expect(wrapper.state()).toEqual(expected);

      wrapper.instance().checkBoxes(mockEvent);
      expect(wrapper.state()).toEqual(changed);

      wrapper.instance().checkBoxes(mockEvent);
      expect(wrapper.state()).toEqual(expected);

    });
  });
});