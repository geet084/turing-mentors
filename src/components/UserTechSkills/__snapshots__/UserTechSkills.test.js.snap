// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTechSkills should match the correct snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserTechSkills
    updateUserInfo={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="pages"
        >
          Mentor Tech skills
        </p>,
        <form>
          <CreateCheckbox
            checkBoxes={[Function]}
            field="ruby"
            name="ruby"
            value={1}
          />
          <CreateCheckbox
            checkBoxes={[Function]}
            field="javascript"
            name="javascript"
            value={2}
          />
          <CreateCheckbox
            checkBoxes={[Function]}
            field="python"
            name="python"
            value={3}
          />
          <CreateCheckbox
            checkBoxes={[Function]}
            field="java"
            name="java"
            value={4}
          />
          <CreateCheckbox
            checkBoxes={[Function]}
            field="elixer"
            name="elixer"
            value={5}
          />
          <CreateCheckbox
            checkBoxes={[Function]}
            field="c"
            name="c"
            value={6}
          />
          <CreateCheckbox
            checkBoxes={[Function]}
            field="php"
            name="php"
            value={7}
          />
          <CreateCheckbox
            checkBoxes={[Function]}
            field="swift"
            name="swift"
            value={8}
          />
          <CreateCheckbox
            checkBoxes={[Function]}
            field="sql"
            name="sql"
            value={9}
          />
        </form>,
        <button
          className="next-btn"
          onClick={[Function]}
        >
          Back
        </button>,
        <button
          className="next-btn"
          onClick={[Function]}
        >
          Next
        </button>,
        <span
          className="pages"
        >
          5 of 
          4
        </span>,
      ],
      "className": "form-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Mentor Tech skills",
          "className": "pages",
        },
        "ref": null,
        "rendered": "Mentor Tech skills",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <CreateCheckbox
              checkBoxes={[Function]}
              field="ruby"
              name="ruby"
              value={1}
            />,
            <CreateCheckbox
              checkBoxes={[Function]}
              field="javascript"
              name="javascript"
              value={2}
            />,
            <CreateCheckbox
              checkBoxes={[Function]}
              field="python"
              name="python"
              value={3}
            />,
            <CreateCheckbox
              checkBoxes={[Function]}
              field="java"
              name="java"
              value={4}
            />,
            <CreateCheckbox
              checkBoxes={[Function]}
              field="elixer"
              name="elixer"
              value={5}
            />,
            <CreateCheckbox
              checkBoxes={[Function]}
              field="c"
              name="c"
              value={6}
            />,
            <CreateCheckbox
              checkBoxes={[Function]}
              field="php"
              name="php"
              value={7}
            />,
            <CreateCheckbox
              checkBoxes={[Function]}
              field="swift"
              name="swift"
              value={8}
            />,
            <CreateCheckbox
              checkBoxes={[Function]}
              field="sql"
              name="sql"
              value={9}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "ruby",
            "nodeType": "class",
            "props": Object {
              "checkBoxes": [Function],
              "field": "ruby",
              "name": "ruby",
              "value": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "javascript",
            "nodeType": "class",
            "props": Object {
              "checkBoxes": [Function],
              "field": "javascript",
              "name": "javascript",
              "value": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "python",
            "nodeType": "class",
            "props": Object {
              "checkBoxes": [Function],
              "field": "python",
              "name": "python",
              "value": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "java",
            "nodeType": "class",
            "props": Object {
              "checkBoxes": [Function],
              "field": "java",
              "name": "java",
              "value": 4,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "elixer",
            "nodeType": "class",
            "props": Object {
              "checkBoxes": [Function],
              "field": "elixer",
              "name": "elixer",
              "value": 5,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "c",
            "nodeType": "class",
            "props": Object {
              "checkBoxes": [Function],
              "field": "c",
              "name": "c",
              "value": 6,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "php",
            "nodeType": "class",
            "props": Object {
              "checkBoxes": [Function],
              "field": "php",
              "name": "php",
              "value": 7,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "swift",
            "nodeType": "class",
            "props": Object {
              "checkBoxes": [Function],
              "field": "swift",
              "name": "swift",
              "value": 8,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "sql",
            "nodeType": "class",
            "props": Object {
              "checkBoxes": [Function],
              "field": "sql",
              "name": "sql",
              "value": 9,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Back",
          "className": "next-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Back",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Next",
          "className": "next-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Next",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "5 of ",
            "4",
          ],
          "className": "pages",
        },
        "ref": null,
        "rendered": Array [
          "5 of ",
          "4",
        ],
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="pages"
          >
            Mentor Tech skills
          </p>,
          <form>
            <CreateCheckbox
              checkBoxes={[Function]}
              field="ruby"
              name="ruby"
              value={1}
            />
            <CreateCheckbox
              checkBoxes={[Function]}
              field="javascript"
              name="javascript"
              value={2}
            />
            <CreateCheckbox
              checkBoxes={[Function]}
              field="python"
              name="python"
              value={3}
            />
            <CreateCheckbox
              checkBoxes={[Function]}
              field="java"
              name="java"
              value={4}
            />
            <CreateCheckbox
              checkBoxes={[Function]}
              field="elixer"
              name="elixer"
              value={5}
            />
            <CreateCheckbox
              checkBoxes={[Function]}
              field="c"
              name="c"
              value={6}
            />
            <CreateCheckbox
              checkBoxes={[Function]}
              field="php"
              name="php"
              value={7}
            />
            <CreateCheckbox
              checkBoxes={[Function]}
              field="swift"
              name="swift"
              value={8}
            />
            <CreateCheckbox
              checkBoxes={[Function]}
              field="sql"
              name="sql"
              value={9}
            />
          </form>,
          <button
            className="next-btn"
            onClick={[Function]}
          >
            Back
          </button>,
          <button
            className="next-btn"
            onClick={[Function]}
          >
            Next
          </button>,
          <span
            className="pages"
          >
            5 of 
            4
          </span>,
        ],
        "className": "form-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Mentor Tech skills",
            "className": "pages",
          },
          "ref": null,
          "rendered": "Mentor Tech skills",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <CreateCheckbox
                checkBoxes={[Function]}
                field="ruby"
                name="ruby"
                value={1}
              />,
              <CreateCheckbox
                checkBoxes={[Function]}
                field="javascript"
                name="javascript"
                value={2}
              />,
              <CreateCheckbox
                checkBoxes={[Function]}
                field="python"
                name="python"
                value={3}
              />,
              <CreateCheckbox
                checkBoxes={[Function]}
                field="java"
                name="java"
                value={4}
              />,
              <CreateCheckbox
                checkBoxes={[Function]}
                field="elixer"
                name="elixer"
                value={5}
              />,
              <CreateCheckbox
                checkBoxes={[Function]}
                field="c"
                name="c"
                value={6}
              />,
              <CreateCheckbox
                checkBoxes={[Function]}
                field="php"
                name="php"
                value={7}
              />,
              <CreateCheckbox
                checkBoxes={[Function]}
                field="swift"
                name="swift"
                value={8}
              />,
              <CreateCheckbox
                checkBoxes={[Function]}
                field="sql"
                name="sql"
                value={9}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "ruby",
              "nodeType": "class",
              "props": Object {
                "checkBoxes": [Function],
                "field": "ruby",
                "name": "ruby",
                "value": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "javascript",
              "nodeType": "class",
              "props": Object {
                "checkBoxes": [Function],
                "field": "javascript",
                "name": "javascript",
                "value": 2,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "python",
              "nodeType": "class",
              "props": Object {
                "checkBoxes": [Function],
                "field": "python",
                "name": "python",
                "value": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "java",
              "nodeType": "class",
              "props": Object {
                "checkBoxes": [Function],
                "field": "java",
                "name": "java",
                "value": 4,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "elixer",
              "nodeType": "class",
              "props": Object {
                "checkBoxes": [Function],
                "field": "elixer",
                "name": "elixer",
                "value": 5,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "c",
              "nodeType": "class",
              "props": Object {
                "checkBoxes": [Function],
                "field": "c",
                "name": "c",
                "value": 6,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "php",
              "nodeType": "class",
              "props": Object {
                "checkBoxes": [Function],
                "field": "php",
                "name": "php",
                "value": 7,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "swift",
              "nodeType": "class",
              "props": Object {
                "checkBoxes": [Function],
                "field": "swift",
                "name": "swift",
                "value": 8,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "sql",
              "nodeType": "class",
              "props": Object {
                "checkBoxes": [Function],
                "field": "sql",
                "name": "sql",
                "value": 9,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Back",
            "className": "next-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Back",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Next",
            "className": "next-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Next",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "5 of ",
              "4",
            ],
            "className": "pages",
          },
          "ref": null,
          "rendered": Array [
            "5 of ",
            "4",
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
